R/junction-read-count.R

```diff
@@ -83,36 +108,37 @@ calculateJunctionReadCounts <- function(exonRanges, intronRanges, junctionFilePa
+#'                                                    genome = NULL
 #'                                                    numberOfCores = 1)
-#' }
+#'                                                    
+#' junctionFilePaths <- c('./sample1.bam', './sample2.bam')
+#' junctionFileLabels <- c('sample1', 'sample2')
+#' promoterReadCounts <- calculatePromoterReadCounts(promoterAnnotationData,
+#'                                                    junctionFilePaths,
+#'                                                    junctionFileLabels,
+#'                                                    junctionType = 'bam',
+#'                                                    genome = 'hg19',
+#'                                                    numberOfCores = 1)}
 #'
 calculatePromoterReadCounts <- function(promoterAnnotationData, junctionFilePaths = NULL, junctionFileLabels = NULL,
-                                        junctionType = 'tophat', numberOfCores = 1) {
-  if (!junctionType %in% c('tophat', 'star')) {
-    stop(paste0('Error: Invalid junction type: ', junctionType, '! Possible values: "tophat" or "star"'))
-  }
-  if (is.null(junctionFilePaths)) {
-    stop(paste0('Error: Please specify valid junction file paths!'))
-  }
-
-  checkFile <- file.exists(junctionFilePaths)
-  if (any(!checkFile)) {
-    stop(paste0('Error: Please specify valid junction file paths. The following file does not exist: ', junctionFilePaths[!checkFile]))
-  }
-
+                                        junctionType = NULL , genome = NULL, numberOfCores = 1) {
+  
   if (numberOfCores == 1) {
     promoterReadCounts <- lapply(junctionFilePaths, calculateJunctionReadCounts, exonRanges = reducedExonRanges(promoterAnnotationData),
-                                 intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType)
+                                 intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType, genome = genome)
   } else {
-    if (requireNamespace('parallel', quietly = TRUE) == TRUE) {
-      promoterReadCounts <- parallel::mclapply(junctionFilePaths, calculateJunctionReadCounts, exonRanges = reducedExonRanges(promoterAnnotationData),
-                                               intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType, mc.cores = numberOfCores)
+    if (requireNamespace('BiocParallel', quietly = TRUE) == TRUE) {
+      bpParameters <- BiocParallel::bpparam()
+      bpParameters$workers <- numberOfCores
+      promoterReadCounts <- BiocParallel::bplapply(junctionFilePaths, calculateJunctionReadCounts, exonRanges = reducedExonRanges(promoterAnnotationData),
+                                               intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType, genome = genome, 
+                                               BPPARAM = bpParameters)
     } else {
```

Comment:
would it make sense to have the following if-else to avoid repeated code? 
if(core>1){
 if(BiocParallel){
bplapply()
} else {
(show warning)
}
} else {
lapply
}

Reply:
if core > 1 and BiocParallel is not available, the warning will be displayed but lapply will not be run?

Reply:
see the change, does that make more sense? it will avoid having the same code line twice